tpke/src/ciphertext.rs

```diff
@@ -120,23 +120,27 @@ pub fn checked_decrypt_with_shared_secret<E: PairingEngine>(
     Some(decrypt_with_shared_secret(ciphertext, s))
 }
 
+fn blake2s_hash(input: &[u8]) -> Vec<u8> {
+    let mut hasher = blake2b_simd::Params::new().hash_length(32).to_state();
+    hasher.update(input);
+    hasher.finalize().as_bytes().to_vec()
+}
+
 pub fn shared_secret_to_chacha<E: PairingEngine>(
     s: &E::Fqk,
 ) -> ChaCha20Poly1305 {
     let mut prf_key = Vec::new();
     s.write(&mut prf_key).unwrap();
-    let mut blake_params = blake2b_simd::Params::new();
-    blake_params.hash_length(32);
-    let mut hasher = blake_params.to_state();
-    prf_key.write(&mut hasher).unwrap();
-    let mut prf_key_32 = [0u8; 32];
-    prf_key_32.clone_from_slice(hasher.finalize().as_bytes());
+    let prf_key_32 = blake2s_hash(&prf_key);
 
     ChaCha20Poly1305::new(GenericArray::from_slice(&prf_key_32))
 }
 
-fn blinded_to_nonce<E: PairingEngine>(nonce: E::G1Affine) -> Nonce {
-    let mut nonce_bytes = Vec::new();
-    nonce.serialize_unchecked(&mut nonce_bytes).unwrap();
-    *Nonce::from_slice(&nonce_bytes[..12])
+fn commitment_to_nonce<E: PairingEngine>(commitment: E::G1Affine) -> Nonce {
```

Comment:
Better `nonce_from_commitment()` so it can't be interpreted that this is a "commitment to a nonce"